Source/WebCore/rendering/updating/RenderTreeBuilderFormControls.cpp

```diff
@@ -99,4 +103,47 @@ RenderBlock& RenderTreeBuilder::FormControls::findOrCreateParentForChild(RenderM
     return *innerRenderer;
 }
 
+void RenderTreeBuilder::FormControls::updateCheckmark(RenderBlockFlow& flow)
+{
+    RefPtr inputElement = dynamicDowncast<HTMLInputElement>(flow.element());
+    ASSERT(inputElement);
+
+    for (CheckedRef child : childrenOfType<RenderElement>(flow)) {
+        if (child->style().pseudoElementType() == PseudoElementType::Checkmark) {
+            if (!inputElement->checked() || flow.style().appearance() != StyleAppearance::Base)
```

Comment:
We should also be generating the checkmark when the input element isn't checked, it's just `visibility: hidden`.

Reply:
I followed what `RenderTreeBuilder::List::updateItemMarker()` does when `style.listStyleType().isNone()` becomes true and the marker was created. It calls `m_builder.destroy(*marker);`